website/src/models/group.js

```diff
@@ -194,7 +194,7 @@ GroupSchema.methods.finishQuest = function(quest, cb) {
 }
 
 function isOnQuest(user,progress,group){
-  return group && progress && user.party.quest.key && group.quest && user.party.quest.key == group.quest.key && group.quest.active;
+  return group && progress && group.quest && group.quest.active && group.quest.members[user._id] == true;
```

Comment:
Let's make this `group.quest.members[user._id] === true`. Otherwise, it seems reasonable to me.